sstar/train.py

```diff
@@ -0,0 +1,796 @@
+        statsmodels_file = output_prefix + ".statsmodels.pickle"
+
+        #call training functions
+        train_statsmodels(train_df, statsmodels_file)
+        train_scikit(train_df, scikit_file)
+
+    return train_df
+
+
+def _train_archie_folders(demo_model_file, nrep, nref, ntgt, ref_id, tgt_id, src_id, seq_len, mut_rate, rec_rate, thread, output_prefix, output_dir, scikit=True, statsmodels=True, drop_dynamic_cols=True, do_training = False):	
+    """
+    Description:
+        compute prdictions for files in all subdirectories of the path indicated by output_dirs
+
+    Arguments:
+        demo_model_file str: demographic model
+        nrep int: number of replicates
+        nref int: number of reference individuals
+        ntgt int: number of target individuals
+        ref_id str: name of reference population
+        tgt_id str: name of target population
+        src_id str: name os fource population
+        seq_len int: sequence length
+        mut_rate float: mutation rate
+        rec_rate float: recombination rate
+        threat int: number of threads
+        output_prefix str: string used to determine logistic regression model name
+        output_dir str: indicates folder with subdirectories containing files for prediction
+        scikit bool: if True, a model using scikit-learn is trained
+        statsmodels bool: if True, a model using statsmodels is trained
+        drop_dynamic_cols bool: if True, features of non-fixed size are removed
+        do_training: if True, models are trained; otherwise only statistics for training are computed
+
+    Returns:
+        train_df DataFrame: contains all windows of train data and corresponding information (statistics, label,...)
+    """
+
+
+    outputfolder = output_dir + "_features_df"
+
+    if not os.path.exists(outputfolder):
+        os.makedirs(outputfolder)    
+
+    #set filenames for individuals, reference and target
+    ref_ind_file = "new_sim" + "_nref" + str(nref) + "_ntgt" + str(ntgt) + ".ref.ind.list"
+    tgt_ind_file  = "new_sim" + "_nref" + str(nref) + "_ntgt" + str(ntgt) + ".tgt.ind.list"
+
+    create_ref_tgt_file(nref, ntgt, ref_ind_file, tgt_ind_file)
+
+    anc_allele_file = None
+
+    #set window length and stepsize
+    win_len = 50000
+    win_step = 50000
+   
+    #I think these parameters are NOT necessary for ArchIE - just retained for the signature of preprocess.process_data
+    match_bonus = 1
+    max_mismatch = 1
+    mismatch_penalty = 1
+    process_archie = True
+
+    #tracts with a proportion between not_archaic and archaic are labeled as ambiguous (in _label)
+    archaic_prop = 0.7
+    not_archaic_prop = 0.3
+
+    true_tracts = []
+    true_tracts_labeled = []
+    features = []
+    file_names = []
+    replicate_counter = 0
+
+    #reading of data, preprocessing - i.e., calculating statistics -, and obtaining & labeling of true tracts
+    for replicate1, folder in enumerate(os.listdir(output_dir)):
```

Comment:
Using `enumerate` makes the order of replicates in the feature table not consistent with the order of the folders containing simulated data

Reply:
This makes it difficult to check the correctness of the calculation for different features.